contracts/zero-ex/contracts/src/fixins/FixinProtocolFees.sol

```diff
@@ -0,0 +1,115 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/IEtherTokenV06.sol";
+import "../external/FeeCollector.sol";
+
+/// @dev Helpers for collecting protocol fees.
+abstract contract FixinProtocolFees {
+    bytes32 immutable feeCollectorCodeHash;
+
+    constructor() internal {
+        feeCollectorCodeHash = keccak256(type(FeeCollector).creationCode);
+    }
+
+    /// @dev   Collect the specified protocol fee in either WETH or ETH. The
+    ///        fee is stored in a per-pool fee collector contract.
+    /// @param poolId The pool ID for which a fee is being collected.
+    /// @param amount The amount of ETH/WETH to be collected.
+    /// @param weth The WETH token contract.
+    function _collectProtocolFee(
+        bytes32 poolId,
+        uint256 amount,
+        IERC20TokenV06 weth
+    )
+        internal
+    {
+        require(
+            msg.value == amount || msg.value == 0,
+            "FixinProtocolFees/WRONG_ETHER_AMOUNT");
+
+        FeeCollector feeCollector = _getFeeCollector(poolId);
+
+        if (msg.value == 0) {
+            // WETH
+            require(weth.transferFrom(msg.sender, address(feeCollector), amount));
+        } else {
+            // ETH
+            (bool success,) = address(feeCollector).call{value: amount}("");
+            require(success);
+        }
+    }
+
+    /// @dev Transfer fees for a given pool to the staking contract.
+    /// @param poolId Identifies the pool whose fees are being paid.
+    function _transferFeesForPool(
```

Comment:
Who will end up calling this?

Reply:
Good question. I think the answer is that we need to expose it publicly and let _anyone_ call it. (0x Labs will likely be responsible for actually doing this at the end of every epoch. No one else has as much incentive to pay the gas fees to do it.)

So I could have perhaps made it an `external`, open function in the fixin itself. But I think it's better to expose it exactly once from a feature.